### DISCUSSION TITLE
manage names of Squeezebox players

### DISCUSSION THREAD

**Discussion by syst3x** [music-assistant/support#2968] [Feature requests and ideas]
I've noticed that Music Assistant sometimes seems to clobber my squeezebox player names. I have 7 hardware players, with names set via LMS, and when I shut down LMS and start up MA, some of these players will have their name changed to "controller:<MAC address>", while others will mostly retain their name, but with the last character removed, while others still will remain correct. 

I started this as a discussion/feature request because I'm not sure there's any claim that MA respects Squeezebox player names, but the very inconsistent behavior makes me wonder what the intent is, and whether a feature could be added to change player names, or at least not clobber them once they've been set via LMS.

(Yes, I'm aware of the "friendly name" option in MA, but that does not change the player name as viewed on the player itself.)

**Comment by MATZE-MAN** [comment]
Hello, I have the same problem. I have been an LMS user for years and currently have two Squeezelite players, a Sonos Ikea speaker and two Logitech radios. When I wanted to add the LMS as a repository to homeassistant yesterday, I saw the new music assistant server. Of course, I tested it straight away. With the Logitech radios, playback sometimes doesn't work. The bar in the web interface runs for about 10 seconds and then starts again from the beginning. The players get their MAC address or the last letter in the player's name is missing. The hardware favorites buttons on the Logitech radio can be assigned via the web interface, but are not transferred to the radio. The log reports an error in the Python script. Unfortunately, the favorites etc. are not displayed for selection on the radios. It is not possible to operate the radio using its buttons. The display shows photos after activation in the web interface. But playback only results in a black screen. I like the design and functionality of the server. Modern and really uses a lot of extensions and player standards. Just like open source projects. 👍 I think I'll post a few feature requests and bugs with logs in the next few days. Really cool project and I hope it will continue to be developed.

**Reply by OzGav** [reply]
Player names you set from LMS are not visible to MA.
You need to set your own names - users should either use LMS or MA, not both

**Reply by syst3x** [reply]
That's surprising to hear. Do you know why MA *sometimes* retains the player names? Name querying is part of the slim protocol and I'm pretty sure is retained on the players themselves...

**Reply by MATZE-MAN** [reply]
ahhh, I took a look at the Slimproto.pm on github. I'm not a pro, there under sub slimproto_accept... the device variable $peer = $clientsock->peeraddr; is assigned. That probably refers to the MAC address of the negotiation with the player, right? I've also just remembered that I had manually assigned the name to the radio (with the name and the missing last letter) via SSH using the hostname file. I had a lot of problems with the radios losing connections and was trying out a few things. It is apparently cached somewhere in the radio and is probably leading to the strange naming in MA. But it is true that at the moment no titles and covers are being transferred to the radios and the favorites etc. are not working, or should they actually?

**Reply by OzGav** [reply]
Cover art is an existing. Issue https://github.com/music-assistant/hass-music-assistant/issues/2483

**Reply by MATZE-MAN** [reply]
ahh thanks for the info 👍 I'll take a look

**Reply by OzGav** [reply]
@syst3x Sorry I do not. The answer I provided was from the devs

**Reply by axodentally** [reply]
did the preset buttons on the radio ever work for you @MATZE-MAN? Since they don't for me and I am wondering whether it is my fault or a bug.

**Reply by axodentally** [reply]
Okay, I do not know what the problem was (the logs did not show anything, when pressing the preset buttons or when changing the preset settings in MASS), maybe a restart of the Squeezebox Radio was the fix, but whatever it was, the presets buttons now work correctly.

Sorry for pinging.
